server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalData.scala

```diff
@@ -41,14 +41,15 @@ class CreateOrUpdateAdditionalData(implicit schema: DBSchema, dbEngine: DoobieEn
     val partitioning = PartitioningForDB.fromSeqPartitionDTO(values.partitioning)
     val partitioningNormalized = SerializationUtils.asJson(partitioning)
 
-     val additionalDataNormalized = SerializationUtils.asJson(values.additionalData)
+    // implicits from Doobie can't handle Map[String, Option[String]] -> HStore, so we converted None to null basically
+    val additionalDataNormalized = values.additionalData.map{ case (k, v) => (k, v.orNull)}
```

Comment:
actually `mapValues` doesn't return `Map` but a `View` :( 

https://stackoverflow.com/questions/14882642/scala-why-mapvalues-produces-a-view-and-is-there-any-stable-alternatives

So it's either something like:

```
    // implicits from Doobie can't handle Map[String, Option[String]] -> HStore, so we converted None to null basically
    // also, mapValues returns a view, not map, so Doobie can't work with it
    // https://stackoverflow.com/questions/14882642/scala-why-mapvalues-produces-a-view-and-is-there-any-stable-alternatives
    val additionalDataNormalized = values.additionalData.mapValues(_.orNull).view.force
```

or my original solution - I'll stick to my original solution because it's more direct and clear